File: DemoProject/DemoProject/SystemForm/FrmPassChange.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Data.SqlClient;
using DemoProject.DAL;
using DemoProject.SO;

namespace DemoProject.SystemForm
{
    public partial class FrmPassChange : Form
    {
        protected void AlignCenterToScreen()
        {
            //grbdetailkhoabomon.Show();
            Screen screen = Screen.FromControl(this);

            Rectangle workingArea = screen.WorkingArea;
            this.Location = new Point()
            {
                X = Math.Max(workingArea.X, workingArea.X + (workingArea.Width - this.Width) / 2),
                Y = Math.Max(workingArea.Y, workingArea.Y + (workingArea.Height - this.Height) / 2)
            };
        }
        public FrmPassChange()
        {
            InitializeComponent();
        }

        private void btnHuyBo_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void btnLuu_Click(object sender, EventArgs e)
        {
            SqlConnection cnn = new SqlConnection(MyApp.MSSQLConnectionString);
            cnn.Open();
            if (txtTenDN.Text == MyApp.gUserDB)
            {
                if (txtMatKhauCu.Text == MyApp.gPwdDB)
                {
                    if (txtMatKhauMoi.Text == txtNhapLai.Text)
                    {
                        string newpass = txtMatKhauMoi.Text;
                        SqlConnection.ChangePassword(MyApp.MSSQLConnectionString, newpass);
                        MyApp.gPwdDB = txtMatKhauMoi.Text;
                        MessageBox.Show("Đổi mật khẩu thành công! ");
                    }
                    else { MessageBox.Show("Lỗi thay đổi mật khẩu!"); }
                }
                else { MessageBox.Show("Mật khẩu cũ không đúng!"); }
            }
            else
            {
                MessageBox.Show("Tên đăng nhập không đúng! ");
            }
        }

        private void FrmPassChange_Load(object sender, EventArgs e)
        {
            AlignCenterToScreen();
            txtTenDN.Text = MyApp.gUserDB;
            txtMatKhauCu.Text = MyApp.gPwdDB;
        }

        private void txtNhapLai_TextChanged(object sender, EventArgs e)
        {
           
        }

        private void ckbcheckpass_CheckedChanged(object sender, EventArgs e)
        {
            if (ckbcheckpass.Checked == false)
            {
                txtMatKhauCu.UseSystemPasswordChar = true;
                txtMatKhauMoi.UseSystemPasswordChar = true;
                txtNhapLai.UseSystemPasswordChar = true;
                //txtnhaplai.UseSystemPasswordChar = false;
            }
            else
            {
                txtMatKhauCu.UseSystemPasswordChar = false;
                txtMatKhauMoi.UseSystemPasswordChar = false;
                txtNhapLai.UseSystemPasswordChar = false;
            }
        }
    }
}

File: DemoProject/DemoProject/SystemForm/FrmPassChange.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports System.Data.SqlClient
Imports DemoProject.DAL
Imports DemoProject.SO
Namespace DemoProject.SystemForm
	Public Partial Class FrmPassChange
		Inherits Form
		Protected Sub AlignCenterToScreen()
			Dim screen As Screen = Screen.FromControl(Me)
			Dim workingArea As Rectangle = screen.WorkingArea
			Me.Location = New Point() With { _
				Key .X = Math.Max(workingArea.X, workingArea.X + (workingArea.Width - Me.Width) / 2), _
				Key .Y = Math.Max(workingArea.Y, workingArea.Y + (workingArea.Height - Me.Height) / 2) _
			}
		End Sub
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnHuyBo_Click(sender As Object, e As EventArgs)
			Me.Close()
		End Sub
		Private Sub btnLuu_Click(sender As Object, e As EventArgs)
			Dim cnn As New SqlConnection(MyApp.MSSQLConnectionString)
			cnn.Open()
			If txtTenDN.Text = MyApp.gUserDB Then
				If txtMatKhauCu.Text = MyApp.gPwdDB Then
					If txtMatKhauMoi.Text = txtNhapLai.Text Then
						Dim newpass As String = txtMatKhauMoi.Text
						SqlConnection.ChangePassword(MyApp.MSSQLConnectionString, newpass)
						MyApp.gPwdDB = txtMatKhauMoi.Text
						MessageBox.Show("Đổi mật khẩu thành công! ")
					Else
						MessageBox.Show("Lỗi thay đổi mật khẩu!")
					End If
				Else
					MessageBox.Show("Mật khẩu cũ không đúng!")
				End If
			Else
				MessageBox.Show("Tên đăng nhập không đúng! ")
			End If
		End Sub
		Private Sub FrmPassChange_Load(sender As Object, e As EventArgs)
			AlignCenterToScreen()
			txtTenDN.Text = MyApp.gUserDB
		End Sub
		Private Sub txtNhapLai_TextChanged(sender As Object, e As EventArgs)
		End Sub
		Private Sub ckbcheckpass_CheckedChanged(sender As Object, e As EventArgs)
			If ckbcheckpass.Checked = True Then
				If txtNhapLai.Text = txtMatKhauMoi.Text Then
					MessageBox.Show("Mật khẩu trùng khớp ")
				Else
					MessageBox.Show("Maatk khẩu không trùng khớp ")
				End If
			End If
		End Sub
	End Class
End Namespace
